### ISSUE TITLE
Better explanation needed for interns in syncing with git

### ISSUE THREAD

**Issue by njanne19** [open-learning-exchange/open-learning-exchange.github.io#1003]
For step 5, I don't think the [Git Repositories](http://open-learning-exchange.github.io/#!./pages/gitandrepositories.md) page is clear enough to new developers on the simple stuff. The page should do a better job at telling the new intern exactly what he/she should do in order to ensure their work is never lost to an out of date repo. A solution to this would be to put in simple instructions and in bold text at the top of the page that 

- __BEFORE WORKING ALL DEVELOPERS SHOULD:__
1. ``git diff``
2. ``git status``
3. ``git fetch upstream``
4. ``git checkout master``
5. ``git merge upstream/master``
6. ``git pull``
7. ``git push``
8. Make new branch and start working

**Comment by aurinsomnia** [comment]
While I think more guidance on Git usage could be helpful for new users, I think that simplifying everything down to just a set of commands to type out means that some users won't understand exactly what they're typing and why. As an intern who was mostly unfamiliar with Git before starting, being forced to do research on what everything did and why I needed it helped me a lot in understanding how Git actually worked, and what I could/should use it for. If you can find a way to help new users understand what they're doing, I'd be more in favor of that over a simple list.

**Comment by njanne19** [comment]
How about instead we had something like this:

*You should know that every time you step away from your desk and come back, other people are working at OLE too! Because of this. It is easy to find that your repository is out of date from the current one online, and if this is true when you start working, you could potentially lose all of your work when it comes time to submit. In order to prevent this from happening, we've devised a set of steps that we'll walk you through in order to make sure you stay updated.*

Every time you are ready to work, before you start, so the following: 

1. Locate to your local save of BeLL using the ''cd'' command on terminal or CMD

2.  First, you should check for changes and see if anything new has been added while you were gone. To do this, first type ''git fetch upstream''. This basically does an update of your project, and it gives the terminal information on the changes that have been made in your absence. Make sure, however, that you are on the master branch, which can be done by typing ''git checkout master''. To print the differences between your local files v.s. the upstream repository (link here), you then should type ''git status''. More documentation on both can be found here (link) 

2.  Even though you have completed step 1, you actually haven't updated your local environment yet. Step 1 allowed the terminal to *notice* the changes made in your absence, but now you can actually download those changes by typing in the terminal ''git merge upstream/master''. This command tells your terminal to download anything you've missed since you last worked. More information on merge can be found here (link). 

3. Finally, you're going to want to push those changes you downloaded to your *own* personal repository. If you recall, your personal repository is named <username>.github.io to make sure your personal repository is updated, just type ''git push'', and the terminal will push all the updates to your local repository. NOTE: on some occasions you might find yourself having to ''git pull'' from your local repository. This usually only happens if it has something that your local repository doesn't have, and this occurs very infrequently. More information on push/pull can be found here(link).

4. Your done that's it! You can start working by making a new branch and having at it. Remember to update your repositories regularly. It will save you a lot of time and effort if you always stay up to date.

**Comment by njanne19** [comment]
Also the your in step 5 should be a you're, I'll make sure the grammar is correct on the real thing

**Comment by adhikara** [comment]
On the contrary, I found the Git page very useful, albeit a little confusing. It may be useful to append these instructions at the END, for people who forgot what to do after what step, but even then it's probably better to see the explanations again.

**Comment by njanne19** [comment]
So do you want me to implement this for the sake of finishing the intern steps? Or do you recommend me going on to trying to make an issue somewhere else? I'm fine with either.

**Comment by navneetkarnam** [comment]
@njanne19 I think you should try to shorten your explanation and then make a PR for this. I feel it is a bit long given that there are instructions on this in the page. Also do specify where are you looking to add this.

**Comment by njanne19** [comment]
@navneetkarnam Thanks for the response, I can definitely shorten this because it was a bit wordy. I was looking to add this at the end of the [Git Repositories](http://open-learning-exchange.github.io/#!./pages/gitandrepositories.md) page as suggested by @adhikara. I could shorten this down to a small summary and add it to the bottom.

**Comment by navneetkarnam** [comment]
@njanne19 perfect! do shorten this and make a PR for it!
